MobileCenter/MobileCenter/Internals/Vendor/Reachability/MS_Reachability.m

```diff
@@ -53,22 +53,34 @@ static void ReachabilityCallback(SCNetworkReachabilityRef target, SCNetworkReach
 
 #pragma mark - Reachability extension
 
-@interface MS_Reachability()
+@interface MS_Reachability ()
 
-@property (nonatomic) SCNetworkReachabilityRef reachabilityRef;
+@property(nonatomic) SCNetworkReachabilityRef reachabilityRef;
 
 @end
 
 #pragma mark - Reachability implementation
 
+/*
+ * Instantiation, deallocation and starting/stopping notifier for reachability instance are enforced to
+ * run in main thread. MS_Reachability is not thread-safe so stopNotifier doesn't properly unschedule jobs
+ * from the loop when it is called from different thread, and this generates unexpected crashes that are caused by
+ * accessing disposed instance especially when reachability is used for local variables.
```

Comment:
There's an "a" missing: "from A different thread" and "accessing A disposed instance".